### ISSUE TITLE
Cannot Load the list of files in Google Drive from org-web.org

### ISSUE THREAD

**Issue by aczzdx** [DanielDe/org-web#73]
Hi.

I was able to use org-web from https://org-web.org/ with Google Drive on Firefox several weeks ago. However, when I tried to sign in today, I found that after I signed in to my Google account, the application cannot load the file list and it went back to the first page (https://org-web.org/). 

I checked the HTTP requests. The status codes for the two HTTP request to Google Drive API are 403 with the following payload:

```{json}
{
 "error": {
  "errors": [
   {
    "domain": "usageLimits",
    "reason": "dailyLimitExceededUnreg",
    "message": "Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.",
    "extendedHelp": "https://code.google.com/apis/console"
   }
  ],
  "code": 403,
  "message": "Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup."
 }
}
```

I am using Firefox 69.0 on Mac OS X 10.14, and I tried to sign in again after clearing the cache, but the problem still happened. Is there something I could do on my local machine to fix this problem?

Thanks in advance.

**Comment by danfan1** [comment]
I am seeing the same error on Chrome (Chrome 77 on Mac OS X 10.14). The error payload is the same as above.

**Comment by munen** [comment]
org-web.net is over the quota of free Google Drive usage.

I'm running my own instance of org-web (actually a fork with quite a few changes/fixes). You guys are welcome to use it, as well:

- The alternative instance: https://organice.200ok.ch/
- The repo so you can see the changes: https://github.com/200ok-ch/organice

Not looking to intruce on the org-web community, though. I only answered because some people couldn't use org-web for 4 days by now.

**Comment by DanielDe** [comment]
Hey sorry for the slow response @aczzdx and @danfan1. 

This certainly looks like some kind of quota overuse, but Google's developer doesn't seem to indicate that there have been any problems. But I'll see if I can raise the limit!

**Comment by aczzdx** [comment]
Hi @DanielDe,

Thanks for your reply and your attention on this issue.

I am looking forward to the raising limit of Google API, but I am also thinking about run org-web on my local machine.

Thanks again for your wonderful project.

**Comment by DanielDe** [comment]
I think that's a great solution @aczzdx. Its pretty straightforward to set up your own Google Drive app and substitute the key in org-web. Let me know if I can help you with that!

**Comment by sai-prasanna** [comment]
I get the same issue now. Is self-hosting the optimal way to run org-web?

**Comment by DanielDe** [comment]
hey @sai-prasanna - yeah, when I reached out about that limit I don't think I actually got a response. I'll follow up on that and see what I can do!

Since Dropbox works fine still (org-web is already a "production" app on Dropbox), another simple solution is to install Dropbox and use it exclusively to sync your org files (via a symlink).

**Comment by DanielDe** [comment]
Hmm, interestingly I'm not showing anything close to hitting the quota on the API console, and I'm able to log in just fine. 

@sai-prasanna are you still experiencing the issue? Also, did you post that comment as soon as you hit it the first time? Knowing exactly when this happened may help me track down the issue.

**Comment by sai-prasanna** [comment]
Yeah, I posted the comment as soon as I and my colleague hit it. It happens in both firefox and chrome.

**Comment by DanielDe** [comment]
@sai-prasanna any chance you'd be willing to hop on a Google Hangout (or similar) to help me debug this? Since I can't seem to reproduce this myself it could be really helpful to see whats happening on your screen. If so, I'll send you some contact info so we can set up a call!

**Comment by JonathanReeve** [comment]
I can confirm that this is happening on my end, as well, for what it's worth. It appears to happen across browsers.

**Comment by DanielDe** [comment]
Ah good to know. Would you be willing to hop on a screen share with me @JonathanReeve to help diagnose the issue? I’d super appreciate it!

**Comment by JonathanReeve** [comment]
Sure.

**Comment by johnhamelink** [comment]
Hey there, I'm also experiencing this issue - is there anything I can do to help debug? I'm running Firefox 77.0.1 on Arch Linux

**Comment by DanielDe** [comment]
> Hey there, I'm also experiencing this issue - is there anything I can do to help debug? I'm running Firefox 77.0.1 on Arch Linux

A quick screenshot of the console in your developer tools would be helpful!

**Comment by johnhamelink** [comment]
Hope this helps!

![Screenshot_2020-06-19_02-05-50](https://user-images.githubusercontent.com/101739/85085995-6d677780-b1d1-11ea-97df-26d8180426dc.png)

**Comment by tbc3** [comment]
Hi all,

any news on this issue? I have it as well.

**Comment by DanielDe** [comment]
Hey @tbc3, unfortunately, no, I haven't been able to figure this one out yet (I also have pretty limited time to spend on this these days). Dropbox still seems to work great, so if possible I'd recommend syncing your org file with Dropbox.

**Comment by tbc3** [comment]
the problem I have, which isn't yours, in dropbox limits me to 3 connections on the free side. I choose google for my online storage so I can connect to several machines. I'll keep watching in case you get time to work on it. Thanks!

**Comment by ngharrison** [comment]
I just found this web app and am interested in its Google Drive support (I have the same issue on my attempts to use it). Since I'm new to this discussion, this may already have been investigated, but I have found something related in my searching. The two answers on to this post appear relevant:

https://stackoverflow.com/questions/12503437/google-drive-service-account-returns-403-usagelimits

Possibly missing authentication information on the server side.

**Comment by DanielDe** [comment]
Thanks for the link, @ngharrison! Dropbox is issuing some update for their API soon, so I'll be taking a look at both syncing solutions again in the near ftureu.

**Comment by ngharrison** [comment]
Sounds great! I can contribute or debug if you can think of a way. It just seemed to me like it mostly has to do with the connected Google API account.
